platform/InternalHttp.roc

```diff
@@ -1,115 +1,139 @@
+Response : {
+    status : U16,
     headers : List Header,
-    url : Str,
-    mimeType : Str,
     body : List U8,
-    timeoutMilliseconds : U64,
 }
 
-fromHostRequest : RequestToAndFromHost -> Request
-fromHostRequest = \{ method, headers, url, mimeType, body, timeoutMilliseconds } -> {
-    method: methodFromStr method,
-    headers,
-    url,
-    mimeType,
-    body,
-    timeout: if timeoutMilliseconds == 0 then NoTimeout else TimeoutMilliseconds timeoutMilliseconds,
-}
+# FOR HOST
 
-# Name is distinguished from the Timeout tag used in Response and Error
-TimeoutConfig : [TimeoutMilliseconds U64, NoTimeout]
-
-Method : [Options, Get, Post, Put, Delete, Head, Trace, Connect, Patch, Extension Str]
-
-methodFromStr : Str -> Method
-methodFromStr = \str ->
-    when str is
-        "OPTIONS" -> Options
-        "GET" -> Get
-        "POST" -> Post
-        "PUT" -> Put
-        "DELETE" -> Delete
-        "HEAD" -> Head
-        "TRACE" -> Trace
-        "CONNECT" -> Connect
-        "PATCH" -> Patch
-        extension -> Extension extension
-
-Header : {
-    name : Str,
-    value : Str,
+RequestToAndFromHost : {
+    method : U64,
+    method_ext : Str,
+    headers : List Header,
+    uri : Str,
+    body : List U8,
+    timeout_ms : U64,
 }
 
-Response : ResponseToHost
-
-ResponseToHost : {
+ResponseToAndFromHost : {
     status : U16,
     headers : List Header,
     body : List U8,
 }
 
-ResponseFromHost : {
-    variant : Str,
-    metadata : Metadata,
-    body : List U8,
+to_host_response : Response -> ResponseToAndFromHost
+to_host_response = \{ status, headers, body } -> {
+    status,
+    headers,
+    body,
 }
 
-Metadata : {
-    url : Str,
-    statusCode : U16,
-    statusText : Str,
-    headers : List Header,
+to_host_request : Request -> RequestToAndFromHost
+to_host_request = \{ method, headers, uri, body, timeout_ms } -> {
+    method: to_host_method method,
+    method_ext: to_host_method_ext method,
+    headers,
+    uri,
+    body,
+    timeout_ms: to_host_timeout timeout_ms,
 }
 
-Error : [
-    BadRequest Str,
-    Timeout U64,
-    NetworkError,
-    BadStatus { code : U16, body : ErrorBody },
-    BadBody Str,
-]
-
-ErrorBody := List U8 implements [
-        Inspect {
-            toInspector: errorBodyToInspector,
-        },
-    ]
-
-errorBodyToInspector : ErrorBody -> _
-errorBodyToInspector = \@ErrorBody val ->
-    Inspect.custom \fmt ->
-        when val |> List.takeFirst 50 |> Str.fromUtf8 is
-            Ok str -> Inspect.apply (Inspect.str str) fmt
-            Err _ -> Inspect.apply (Inspect.str "Invalid UTF-8 data") fmt
-
-errorBodyToUtf8 : ErrorBody -> List U8
-errorBodyToUtf8 = \@ErrorBody body -> body
+to_host_method : Method -> _
+to_host_method = \method ->
+    when method is
+        OPTIONS -> 5
+        GET -> 3
+        POST -> 7
+        PUT -> 8
+        DELETE -> 1
+        HEAD -> 4
+        TRACE -> 9
+        CONNECT -> 0
+        PATCH -> 6
+        EXTENSION _ -> 2
```

Comment:
I did have a proper rust `Method` impl, but we were having segafults during the development of this upgrade and suspected the Method to be an issue (possibly alignment or refcounting). Turned out not to be the source of our issues, but I decided it was easier just to leave it like this and not re-generate glue. 

The delta is the space difference between a `u8` and `u64` per request.

Reply:
Due to padding I think there literally is no difference.